### PR TITLE
Import Seed cli commands 

### PR DESCRIPTION
**Pull request by benthecarman**
Built on top of #2372 

Dedicated docs will go in a follow up PR with further info like restoring a wallet

### CODE REVIEW

File: app-commons/src/main/scala/org/bitcoins/commons/serializers/Picklers.scala [hunk 1]
```diff
@@ -507,4 +507,12 @@ object Picklers {
...
     }
   }
+
+  implicit val mnemonicCodePickler: ReadWriter[MnemonicCode] =
+    readwriter[String].bimap(
+      _.words.mkString(" "),
+      str => MnemonicCode.fromWords(str.split(' ').toVector))
+
+  implicit val extPrivateKeyPickler: ReadWriter[ExtPrivateKey] =
+    readwriter[String].bimap(_.toString, ExtPrivateKey.fromString)
```
Comment (Christewart) [inline app-commons/src/main/scala/org/bitcoins/commons/serializers/Picklers.scala]: is this going to be a problem calling `_.toString` ? The `ExtPrivateKey extends MaskedToString`

File: app/server/src/main/scala/org/bitcoins/server/WalletRoutes.scala [hunk 2]
```diff
@@ -454,6 +458,56 @@ case class WalletRoutes(wallet: AnyHDWalletApi)(implicit
...
+    case ServerCommand("importxprv", arr) =>
+      ImportXprv.fromJsArr(arr) match {
+        case Failure(err) =>
+          reject(ValidationRejection("failure", Some(err)))
+        case Success(ImportXprv(walletName, xprv, passwordOpt)) =>
+          complete {
+            val seedPath = kmConf.seedFolder.resolve(
+              s"$walletName-${WalletStorage.ENCRYPTED_SEED_FILE_NAME}")
+
+            // TODO Maybe rethink creation time
```
Comment (Christewart) [inline app/server/src/main/scala/org/bitcoins/server/WalletRoutes.scala]: what is this comment for?

File: docs/applications/server.md [hunk 3]
```diff
@@ -187,10 +187,18 @@ For more information on how to use our built in `cli` to interact with the serve
  - `lockunspent` `unlock` `transactions` - Temporarily lock (unlock=false) or unlock (unlock=true) specified transaction outputs.
     - `unlock` - Whether to unlock (true) or lock (false) the specified transactions
     - `transactions` - The transaction outpoints to unlock/lock
- - `walletpassphrasechange` `oldpassphrase` `newpassphrase` - Changes the wallet passphrase
+- `importseed` `walletname` `words` `passphrase` - Imports a mnemonic seed as a new seed file
+   - `walletname` - Name to associate with this seed
+   - `words` - Mnemonic seed words
```
Comment (Christewart) [inline docs/applications/server.md]: what format is this suppose to be ?

Comment (Christewart) [reply docs/applications/server.md]: it's worth adding the answer in the documentation IMO

File: app/server/src/main/scala/org/bitcoins/server/WalletRoutes.scala [hunk 4]
```diff
@@ -454,6 +458,56 @@ case class WalletRoutes(wallet: AnyHDWalletApi)(implicit
...
+          }
+      }
+
+    case ServerCommand("importseed", arr) =>
+      ImportSeed.fromJsArr(arr) match {
+        case Failure(err) =>
+          reject(ValidationRejection("failure", Some(err)))
+        case Success(ImportSeed(walletName, mnemonic, passwordOpt)) =>
+          complete {
+            val seedPath = kmConf.seedFolder.resolve(
```
Comment (Christewart) [inline app/server/src/main/scala/org/bitcoins/server/WalletRoutes.scala:471]: can some of this core logic be moved into the key manager project? It seems out of place being in here

**Review by Christewart — CHANGES_REQUESTED** [review]

File: app/server/src/main/scala/org/bitcoins/server/WalletRoutes.scala (hunk 4 above)
Comment (benthecarman) [reply app/server/src/main/scala/org/bitcoins/server/WalletRoutes.scala:471]: Not really, it's just making a the seed path and the putting the seed/xprv into our data types

**Review by Christewart — APPROVED** [review]

**Pull request merged**